compiler/rustc_resolve/messages.ftl

```diff
@@ -461,7 +461,10 @@ resolve_unexpected_res_use_at_op_in_slice_pat_with_range_sugg =
     if you meant to collect the rest of the slice in `{$ident}`, use the at operator
 
 resolve_unnamed_crate_root_import =
-    crate root imports need to be explicitly named: `use crate as name;`
+    {$dollar ->
+        [true] crate root imports need to be explicitly named: `use $crate as name;`
```

Comment:
I don't think this is a valid suggestion. Per [this](https://github.com/rust-lang/rust/blob/15283f6fe95e5b604273d13a428bab5fc0788f5a/compiler/rustc_resolve/src/build_reduced_graph.rs#L628-L651), `use $crate as x;` is not allowed.

Reply:
More than a partial fixup, I'd rather want to see an exhaustive test listing the possible cases of imports using only path segment keywords (`self`, `super`, `crate`, `$crate`) in the input path with FIXMEs as a declaration of intent.